models/intermediate/int_xero__journal_line_pivoted_tracking_categories.sql

```diff
@@ -0,0 +1,49 @@
+{{ config(enabled=(var('xero__using_journal_line_tracking_category', True)
+        and var('xero__using_tracking_categories', True))) }}
+
+with journal_line_has_tracking as (
+
+    select *
+    from {{ var('journal_line_has_tracking_category') }}
+
+), tracking_categories_with_options as (
+
+    select *
+    from {{ ref('int_xero__tracking_categories_with_options') }}
+
+), journal_tracking as (
+
+    select
+        journal_line_has_tracking.journal_id,
+        journal_line_has_tracking.journal_line_id,
+        journal_line_has_tracking.source_relation,
+        tracking_categories_with_options.tracking_category_name,
+        tracking_categories_with_options.tracking_option_name
+    from journal_line_has_tracking
+
+    left join tracking_categories_with_options
+        on journal_line_has_tracking.tracking_category_id = tracking_categories_with_options.tracking_category_id
+        and journal_line_has_tracking.tracking_category_option_id = tracking_categories_with_options.tracking_option_id
+        and journal_line_has_tracking.source_relation = tracking_categories_with_options.source_relation
+),
+
+final as (
+
+    select
+        journal_id,
+        journal_line_id,
+        source_relation,
+        {{ dbt_utils.pivot(
+            column='tracking_category_name',
+            values=dbt_utils.get_column_values(ref('stg_xero__tracking_category'), 'tracking_category_name'),
```

Comment:
Why is this referencing the staging model? I would assume we would need to reference the `int_xero__tracking_categories_with_options`. Otherwise when pivoting out these fields we could be pivoting out deleted columns.

Reply:
If we make the above update, we'll need to make sure we materialize the `int_xero__tracking_categories_with_options` model as a table or view because I don't believe the pivot macro works on an ephemeral model.

Reply:
If we make the above update, we'll need to make sure we materialize the `int_xero__tracking_categories_with_options` model as a table or view because I don't believe the pivot macro works on an ephemeral model.

Reply:
Updated.